### PR TITLE
add 4 more CSJ models

### PR DESCRIPTION
**Pull request by martinjankowiak**
follow-up to #1809 

## Tasks
- [x] add models to tests/test_examples.rst
- [x] add tutorial/source/cjs.rst

### CODE REVIEW

**Comment by martinjankowiak** [comment]
@fritzo thanks! this should be good to go now (with csj.rst)

**Comment by martinjankowiak** [comment]
@fritzo let me know if you want me to walk you through the new model variants

File: examples/cjs/cjs.py [hunk 1]
```diff
@@ -57,24 +65,184 @@ def model(data):
...
                 mu_y_t = rho * z
                 pyro.sample("y_{}".format(t), dist.Bernoulli(mu_y_t),
-                            obs=data[:, t])
-            first_capture_mask |= data[:, t].byte()
+                            obs=capture_history[:, t])
+            first_capture_mask |= capture_history[:, t].byte()
+
+
+"""
+In our second model variant there is a survival probability phi_t for 6
```
Comment (fritzo) [inline examples/cjs/cjs.py]: nit: consider saying "there is a time-varying survival probability"

File: examples/cjs/cjs.py [hunk 2]
```diff
@@ -57,24 +65,184 @@ def model(data):
...
+
+
+def model_2(capture_history, sex):
+    N, T = capture_history.shape
+    rho = pyro.sample("rho", dist.Uniform(0.0, 1.0))  # recapture probability
+
+    z = torch.ones(N)
+    first_capture_mask = torch.zeros(N).byte()
+    # we create the plate once, outside of the loop over t
+    dippers_plate = pyro.plate("dippers", N, dim=-1)
```
Comment (fritzo) [inline examples/cjs/cjs.py]: What is a dipper?

File: examples/cjs/cjs.py [hunk 3]
```diff
@@ -57,24 +65,184 @@ def model(data):
...
+"""
+In our third model variant there is a survival probability phi_t for 6
+of the 7 years of the capture data (just like in model_2), but here
+each phi_t is treated as a random effect.
+"""
+
+
+def model_3(capture_history, sex):
+    def logit(p):
+        return torch.log(p / (1.0 - p))
```
Comment (fritzo) [inline examples/cjs/cjs.py]: nit: for numerical stability use `return torch.log(p) - torch.log1p(-p)`

File: tutorial/source/cjs.rst [hunk 4]
```diff
@@ -0,0 +1,11 @@
+Capture-Recapture Models (CJS Models)
+===================
```
Comment (fritzo) [inline tutorial/source/cjs.rst]: Does this underbar need to be full length? I've seen sphinx complain before.

File: examples/cjs/cjs.py (hunk 2 above)
Comment (martinjankowiak) [reply examples/cjs/cjs.py]: see top of file: it's norway's national bird

File: examples/cjs/cjs.py (hunk 3 above)
Comment (martinjankowiak) [reply examples/cjs/cjs.py]: done

File: examples/cjs/cjs.py (hunk 1 above)
Comment (martinjankowiak) [reply examples/cjs/cjs.py]: done

File: tutorial/source/cjs.rst (hunk 4 above)
Comment (martinjankowiak) [reply tutorial/source/cjs.rst]: done

File: tests/test_examples.py [hunk 5]
```diff
@@ -19,7 +19,11 @@
     'baseball.py --num-samples=200 --warmup-steps=100 --num-chains=2',
     'lkj.py --n=50 --num-chains=1 --warmup-steps=100 --num-samples=200',
     'bayesian_regression.py --num-epochs=1',
-    'cjs/cjs.py --num-steps=1',
+    'cjs/cjs.py --num-steps=1 -m 1',
```
Comment (fritzo) [inline tests/test_examples.py]: rename to `capture_recapture/cjs.py`

**Pull request merged**